contracts/basic/spread/SpreadGenOne.sol

```diff
@@ -1,58 +1,19 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity 0.8.20;
 
-import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
-import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
-import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
-
-import "@openzeppelin/contracts/utils/Strings.sol";
 import "./DemandSpreadLibsGenOne.sol";
 import "./SpreadStorageLibsGenOne.sol";
 import "./OfferedRateCalculationLibsGenOne.sol";
-import "../../interfaces/IIporContractCommonGov.sol";
-import "../../interfaces/IProxyImplementation.sol";
-import "../../interfaces/types/IporTypes.sol";
-import "../../interfaces/types/AmmTypes.sol";
-import "../../security/IporOwnableUpgradeable.sol";
-import "../../security/PauseManager.sol";
+import "../../security/IporOwnable.sol";
 import "../../libraries/IporContractValidator.sol";
-import "../../amm/libraries/types/AmmInternalTypes.sol";
 import "../../amm/spread/SpreadStorageLibs.sol";
 import "../../basic/interfaces/IAmmStorageGenOne.sol";
 import "../../amm/spread/CalculateTimeWeightedNotionalLibs.sol";
 import "../../amm/libraries/IporSwapLogic.sol";
+import "./ISpreadGenOne.sol";
 
-/// @dev This contract cannot be used directly, should be used only through SpreadRouter.
-contract SpreadGenOne is
-    Initializable,
-    PausableUpgradeable,
-    UUPSUpgradeable,
-    IporOwnableUpgradeable,
-    IProxyImplementation,
-    IIporContractCommonGov
-{
-    struct SpreadInputs {
-        //// @notice Swap's assets DAI/USDC/USDT/stETH/etc.
-        address asset;
-        /// @notice Swap's notional value
-        uint256 swapNotional;
-        /// @notice demand spread factor used in demand spread calculation
-        uint256 demandSpreadFactor;
-        /// @notice Base spread
-        int256 baseSpreadPerLeg;
-        /// @notice Swap's balance for Pay Fixed leg
-        uint256 totalCollateralPayFixed;
-        /// @notice Swap's balance for Receive Fixed leg
-        uint256 totalCollateralReceiveFixed;
-        /// @notice Liquidity Pool's Balance
-        uint256 liquidityPoolBalance;
-        /// @notice Ipor index value at the time of swap creation
-        uint256 iporIndexValue;
-        /// @notice fixed rate cap for given leg for offered rate without demandSpread in 18 decimals
-        uint256 fixedRateCapPerLeg;
-        /// @notice Swap's tenor
-        IporTypes.SwapTenor tenor;
-    }
+/// @dev This contract cannot be used directly, should be used only through Router.
```

Comment:
Remove that message

Reply:
Put here info that spread stores information for all tenors for one particular asset